tests/unit/test_extractor.py

```diff
@@ -0,0 +1,125 @@
+import pytest
```

Comment:
the overview mentions checkpointing. I can't tell if this is tested (might speak to readability of the tests). I don't see a test that stops and restarts extraction where checkpointing would come into play

Reply:
had a discussion and we agreed that we can't really test the checkpointing alone and would more align with integration level testing

Reply:
By "checkpointing" we meant when reading ts data from Fabric, we will use the state store to check if we've encountered this data already or it's new data that needs to be processed. It does not directly relate to service stop and restart, which will be addressed in the integration test.